### PR TITLE
fix(pragma-cli): read ds:usage in block lookup, the vocabulary the graph has

### PR DESCRIPTION
**Pull request by advl**
## Done

**`block lookup` — the flagship read — has been silently omitting its usage narrative for every block on every install.**

The story reads `ds:whenToUse` and `ds:whenNotToUse`. The shipped graph contains **zero** of either. `ds:usage` — which the embedded ontology's own `skos:definition` says *"Subsumes the former whenToUse/whenNotToUse"* — is asserted on **264 of 264 blocks**. A comment in `pragma.conf.ts` rationalised this as *"a graph lacking whenToUse/whenNotToUse degrades gracefully"*; it was describing 100% of production as an edge case.

No test caught it because `blockGraph.ts` still authored the retired vocabulary, and said so in its own docblock. The fixture spoke a dialect production had already left.

- **`pragma.conf.ts`** — the two retired sections replaced by one `usage` / `ds:usage` / "Usage" section at `detailed`. The "degrades gracefully" comment replaced with the actual history.
- **`testing/fixtures/blockGraph.ts`** — authors `ds:usage`. Button carries a non-empty literal with its own sub-headings; **Modal carries `ds:usage ""`** so the fixture covers both shapes that exist in production. Only this property changed; nothing else reshaped.
- **`capabilities/block.parity.test.ts`** — asserts `### Usage`, the demoted `#### When to use`, and that Modal's empty literal renders no heading.
- **New `capabilities/block.shipped.exec.test.ts`** (PROTECTED) — boots the *shipped* pack, per the `examples.exec.test.ts` precedent, so this class of drift is caught against real data rather than a fixture.

### Two facts from the census that changed the design

Measured by parsing the 9,068-quad literal out of `pack.generated.ts`, not assumed:

- **138 of the 264 `ds:usage` literals are empty strings** — only 126 carry prose. An unconditional section would have printed a bare heading for over half of production.
- The non-empty literals **already contain their own `### When to use` / `### When not to use` sub-headings** (78 and 71 of 126).

Both are handled by renderer behaviour that already existed and was verified rather than added: `nestHeadings` demotes the body's own ATX headings under `### Usage`, and `isEmptyValue` skips empty sections.

Also: `ds:usage` is on **seven** block classes, not the four in the original report — Component 110, Subcomponent 86, Pattern 42, Layout 14, plus Group 9, View 2, Variant 1.

### The retired properties are dropped, not kept as a fallback

Justified in `pragma.conf.ts` itself. (1) `ds:usage` subsumes them by definition, so a pack carrying both would print the same guidance twice. (2) No fixture would author them, so they would be two more never-exercised declarations behind the same comment that hid the first two — untestable config is exactly what let this ship. (3) The new guard would need an exemption for them, i.e. the guard disbelieving its own rule.

## QA

1. Built CLI against the **shipped** pack, `global.component.button`:
   - **before** — `### Summary` straight to `### Guidelines`
   - **after** — a full `### Usage` section with the real narrative, and `#### When to use` / `#### When not to use` / `#### Call to actions` nested one level below it
2. `bunx vitest run src/capabilities/block.parity.test.ts src/capabilities/block.shipped.exec.test.ts` → green, including the empty-literal case.
3. `bun run gen:reference` → **no diff**.
4. `bun run check` → clean at package and repo root; block, reference, surface and distribution suites 64/64.

Full-suite failures are confined to `create/*` and other subprocess-spawning tests hitting the default 5,000 ms timeout; they vary run to run, pass in isolation (136/136 for `create`), and touch nothing here. **Worth knowing:** when those tests time out mid-run they leak scaffolded files into `packages/cli/pragma/src/lib/`, which then breaks the *next* run's `tsc` gate with a confusing `Cannot find module './Button.svelte'`. `rm -rf packages/cli/pragma/src/lib` clears it.

### Found, deliberately not fixed

**`ds:figmaLink` is declared on `ds:UIBlock` but asserted on zero blocks** in the shipped pack. The guard checks the ontology *definition* rather than instance data precisely to separate the two cases: a property the ontology never defines is a story bug that can never render (this defect), while one it defines that no instance asserts is an upstream content gap that genuinely does degrade gracefully.

### PR readiness check

- [x] PR has a label: `Bug 🐛`, plus `no visual change`.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] The code follows the appropriate [code standards](https://github.com/canonical/web-code-standards).
- [x] All packages define the required scripts in `package.json` — no manifests are touched.
- [x] New package: **n/a** — none introduced.
- [x] Does not require visual testing — CLI story configuration, fixtures and tests only.

🤖 Generated with [Claude Code](https://claude.ai/code)

https://claude.ai/code/session_011B8Z3Lq1eARN1wLfKGvzqC

### CODE REVIEW

File: packages/cli/pragma/src/capabilities/block.shipped.exec.test.ts [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  })),
+  ...(blockLookup.sections ?? []).map((s) => ({
+    what: `section "${s.name}"`,
+    property: s.property,
+  })),
+  ...(blockLookup.expand ?? []).map((e) => ({
+    what: `expand "${e.name}"`,
+    property: e.relation,
+  })),
+];
```
[the remark this answers is not in the document]
Comment (advl) [reply packages/cli/pragma/src/capabilities/block.shipped.exec.test.ts]: Correct, and fixed in 02b011c — thank you. The guard existed because a story read a term the graph had retired, and enumerating only the top level reproduced that blind spot exactly one level down.

`DECLARED_PROPERTIES` is now WALKED rather than listed: `lookup.by`, then every `expand.select` entry recursively via a `selectedTerms(site, select)` helper that recurses through `isNestedExpand` entries into their own `select`. Ten declaration sites became nineteen — the additions are `ds:name` (as `lookup.by` and at five select sites), `ds:hasModifier`, `ds:propertyType` and `ds:optional`. All nineteen pass against the shipped ontology today. Each case is keyed on the declaration SITE, not the distinct term, so a failure names where it was declared (`expand "properties" → field "type"`), and a term added to any `select` in future is covered by the author who declares it, without their help.

Proof it now catches what it could not: renaming `ds:propertyType` to `ds:propertyTypeX` in `pragma.conf.ts` fails the widened guard —

    × 'expand "properties" → field "type"' reads 'ds:propertyTypeX', which the shipped ontology defines
      → expand "properties" → field "type" reads ds:propertyTypeX, which the shipped
        ontology does not define — it can never render, for any block, on any install

— while the previous version of this file passed all 15 tests green under the same rename. Note that every rendering assertion in the file also stayed green under it, which is the silent failure mode in miniature. (Reverted; not committed.)

On line 148: I checked it and the concern does not carry there in the same form. The only story-derived value in that block is `TYPE_VALUES`, which already reads `lookup.types` wholesale, so it has no omission to close. The one literal list there is the retired-terms pin `["ds:whenToUse", "ds:whenNotToUse"]`, and that is a NEGATIVE assertion — it names terms that must not appear anywhere in the graph — so it cannot silently go stale by omission the way a positive enumeration can; a term missing from it is simply a term not pinned, not a green guard over a broken read. If you meant something else at that line, say so and I will take another look.

File: packages/cli/pragma/pragma.conf.ts [hunk 2]
```diff
@@ -97,8 +97,22 @@ const designSystemStories: readonly PackDefinition[] = [
...
+  // that property's own skos:definition. The shipped pack asserts `ds:usage` on
+  // all 264 blocks and neither retired term on any of them, so every
+  // `block lookup` on every install rendered no usage narrative at all —
+  // silently, because a name that maps onto no schema field is exactly the case
+  // OPTIONAL parity swallows. The retired terms are NOT kept as a fallback:
+  // `ds:usage` subsumes them, so a pack carrying both would print the same
+  // guidance twice, and a declaration no shipped graph can satisfy is
+  // untestable by construction — which is what let the silence ship.
+  // `block.shipped.exec.test.ts` now holds every property declared here to
+  // being asserted in the SHIPPED graph.
```
[the remark this answers is not in the document]
Comment (advl) [reply packages/cli/pragma/pragma.conf.ts]: Correct — the prose overstated the guard, and it is fixed in 02b011c.

The distinction you name is exactly the one the test file is built on and the one that sentence flattened: a term the ontology never DEFINES is a story bug that can never render, on any block, on any install (this defect); a term it defines that no instance ASSERTS yet — `ds:figmaLink` today — is an upstream content gap the OPTIONAL parity degrades over gracefully, and failing on it would be a false positive. Saying "asserted" invited a later reader to strengthen the guard into precisely that.

The comment now reads:

> `block.shipped.exec.test.ts` now holds every graph term declared here — the identity property, fields, sections, expand relations, and every term selected beneath them, walked recursively — to being DEFINED as a property by the SHIPPED ontology. Defined, not asserted, and the difference is the whole point: a term the ontology never defines is a story bug that can never render, on any block, on any install — this defect. A term it defines that no instance asserts yet (`ds:figmaLink` today) is a content gap upstream, which the OPTIONAL parity above degrades over gracefully and which this pack has no business failing on. `ds:usage` alone is ALSO held to being asserted, on every block, because its absence is what shipped.

The last sentence keeps the one genuine instance-level claim scoped to the property it is true of, rather than leaving it generalised across all of them.
